lib/jwt/jwk/rsa.rb

```diff
@@ -28,45 +28,68 @@ def kid
         OpenSSL::Digest::SHA256.hexdigest(sequence.to_der)
       end
 
-      def export
-        {
+      def export(options = {})
+        ret = {
           kty: KTY,
           n: encode_open_ssl_bn(public_key.n),
           e: encode_open_ssl_bn(public_key.e),
           kid: kid
         }
+
+        return ret if options[:include_private] != true
```

Comment:
Also check if the keypair has a private key.

`return ret unless private? && options[:include_private] == true`

Reply:
Good point. Will get back to this when I have the time.